service/matching/matchingEngine.go

```diff
@@ -721,6 +727,10 @@ func (e *matchingEngineImpl) unloadTaskQueue(id *taskQueueID) {
 	}
 }
 
+func (e *matchingEngineImpl) requestUnload(id *taskQueueID) {
+	e.tqMonitor.Signal(tqmonUsurped{id: id})
```

Comment:
the signal should be internal to tqMonitor. Here we should just call e.tqMonitor.UnloadTaskQueue(id). (Or tqMonitor.RequestUnloadTaskQueue() if it is async.)